From 4b1f0e0121728640f01a7eaa0799d06c2a5bef68 Mon Sep 17 00:00:00 2001
From: Paresh Purabhiya <ppurab@codeaurora.org>
Date: Wed, 20 Dec 2017 20:25:15 +0530
Subject: [PATCH 17/17] sepolicy : set write permissions for sysfs_boot_adsp.

allow sysfs_boot_adsp write permissions to
/sys/kernel/boot_adsp/boot node.

Change-Id: I370c6be54b0cad987fb679b66c3d8f8552c3c99a
---
 common/file.te       | 4 ++++
 common/file_contexts | 1 +
 common/init.te       | 2 ++
 3 files changed, 7 insertions(+)

diff --git a/common/file.te b/common/file.te
index ee9d6b9..781409d 100644
--- a/common/file.te
+++ b/common/file.te
@@ -323,6 +323,10 @@ type sysfs_msm_power, fs_type, sysfs_type;
 type qvop-daemon_data_file, file_type, data_file_type;
 
 type sysfs_fm, sysfs_type, fs_type;
+
+# for adsp to load /sys/kernel/b ot_adsp/boot
+type sysfs_boot_adsp, sysfs_type, fs_type;
+
 # /data/vendor/tombstones/ file
 type vendor_tombstone_data_file, file_type, data_file_type;
 
diff --git a/common/file_contexts b/common/file_contexts
index 5585caa..844cab5 100644
--- a/common/file_contexts
+++ b/common/file_contexts
@@ -505,6 +505,7 @@
 /sys/module/app_setting/parameters/lib_name                         u:object_r:sysfs_lib:s0
 /sys/module/process_reclaim(/.*)?  u:object_r:sysfs_process_reclaim:s0
 /sys/module/qpnp_rtc/parameters/poweron_alarm                       u:object_r:sysfs_poweron_alarm:s0
+/sys/kernel/boot_adsp/boot                                          u:object_r:sysfs_boot_adsp:s0
 
 ###################################
 # data files
diff --git a/common/init.te b/common/init.te
index f9782d7..ab6511a 100644
--- a/common/init.te
+++ b/common/init.te
@@ -32,6 +32,8 @@ allow init configfs:lnk_file create_file_perms;
 allow init firmware_file:dir { mounton };
 allow init bt_firmware_file:dir { mounton };
 
+allow init sysfs_boot_adsp:file w_file_perms;
+
 #dontaudit non configfs usb denials
 dontaudit init sysfs:dir write;
 
-- 
2.17.1

